utils/chunked_vector.hh

```diff
@@ -431,18 +424,38 @@ chunked_vector<T, max_contiguous_allocation>::make_room(size_t n, bool stop_afte
 }
 
 template <typename T, size_t max_contiguous_allocation>
-void
-chunked_vector<T, max_contiguous_allocation>::do_reserve_for_push_back() {
-    if (_capacity == 0) {
-        // allocate a bit of room in case utilization will be low
-        reserve(boost::algorithm::clamp(512 / sizeof(T), 1, max_chunk_capacity()));
-    } else if (_capacity < max_chunk_capacity() / 2) {
-        // exponential increase when only one chunk to reduce copying
-        reserve(_capacity * 2);
+template <typename... Args>
+T&
+chunked_vector<T, max_contiguous_allocation>::reserve_for_emplace_back(Args&&... args) {
```

Comment:
The function is misnamed, and will be duplicated for every Args combination.

Suggest to split into prepare_emplace_back() and complete_emplace_back(), then emplace_back() can call prepare(), emplace stuff, then complete().

Reply:
Yes, that's doable.

Reply:
done in v2